apollo-router/src/plugins/authentication/jwks.rs

```diff
@@ -617,40 +617,44 @@ pub(super) fn decode_jwt(
 }
 
 pub(crate) fn jwt_expires_in(context: &Context) -> Duration {
-    let claims = context
+    context
         .get(APOLLO_AUTHENTICATION_JWT_CLAIMS)
-        .map_err(|err| tracing::error!("could not read JWT claims: {err}"))
-        .ok()
-        .flatten();
-    let ts_opt = claims
-        .as_ref()
-        .and_then(|x: &serde_json::Value| match x.as_object() {
-            Some(claims) => claims.get("exp")?.as_i64(),
-            None => {
-                tracing::error!("expected JWT claims to be an object");
-                None
+        .unwrap_or_else(|err| {
+            tracing::error!("could not read JWT claims: {err}");
+            None
+        })
+        .flatten()
+        .and_then(|claims_value: Option<serde_json::Value>| {
+            let claims_obj = claims_value.as_ref()?.as_object();
+            // Extract the expiry claim from the JWT, returning immediately if it's not present
```

Comment:
this seems a little untrue since we're not returning immediately, just propagating `None`, maybe we can just omit that?
```suggestion
            // Extract the expiry claim from the JWT
```